Initial testing bench contract

DOBEN opened the pull request:
## Purpose

addresses #https://github.com/Concordium/concordium-misc-tools/issues/53

## Changes

Add a smart contract that can be used for testing various interaction scenarios with wallets via a front end.

abizjak commented on examples/smart-contract-test-bench/src/lib.rs:
```diff
@@ -0,0 +1,831 @@
+//! Smart contract test bench
```
Maybe provide a bit more context here. That is, the reason for this contract's existence.

abizjak commented on examples/smart-contract-test-bench/src/lib.rs:
```diff
@@ -0,0 +1,831 @@
+#[receive(
+    contract = "smart_contract_test_bench",
+    name = "set_object",
+    parameter = "State",
```
Nice :)

DOBEN commented:
Closing this issue:

Moved the smart contract into this PR and addressed the given comments in that PR:
https://github.com/Concordium/concordium-misc-tools/pull/55
